v3/src/components/graph/components/dot-chart-bars.tsx

```diff
@@ -135,20 +133,33 @@ export const DotChartBars = observer(function DotChartBars({ abovePointsGroupRef
                 // Create a map of cases grouped by legend value so we don't need to filter all cases per value when
                 // creating the bar covers.
                 const caseGroups = new Map()
-                dataset?.cases.forEach(aCase => {
-                  const legendValue = dataset?.getStrValue(aCase.__id__, legendAttrID)
-                  const primaryValue = dataset?.getStrValue(aCase.__id__, dataConfig.attributeID(primaryAttrRole))
+                dataConfig.caseDataArray.forEach(aCase => {
+                  const legendValue = dataset?.getStrValue(aCase.caseID, legendAttrID)
+                  const primaryValue = dataset?.getStrValue(aCase.caseID, dataConfig.attributeID(primaryAttrRole))
                   const primarySplitValue =
-                    dataset?.getStrValue(aCase.__id__, dataConfig.attributeID(primarySplitAttrRole))
+                    dataset?.getStrValue(aCase.caseID, dataConfig.attributeID(primarySplitAttrRole))
                   const secondarySplitValue =
-                    dataset?.getStrValue(aCase.__id__, dataConfig.attributeID(secondarySplitAttrRole))
+                    dataset?.getStrValue(aCase.caseID, dataConfig.attributeID(secondarySplitAttrRole))
                   const caseGroupKey =
                     `${legendValue}-${primaryValue}-${primarySplitValue}-${secondarySplitValue}`
                   if (!caseGroups.has(caseGroupKey)) {
                     caseGroups.set(caseGroupKey, [])
                   }
                   caseGroups.get(caseGroupKey).push(aCase)
                 })
+
+                // If the legend attribute is numeric, sort legendCats in descending order making sure to handle any NaN
+                // values for cases that don't have a numeric value for the legend attribute.
+                if (dataConfig.attributeType("legend") === "numeric") {
+                  legendCats.sort((cat1: string, cat2: string) => {
```

Comment:
With the changes to DataConfigurationModel, this sorting is redundant, right? (I tried commenting it out and bar chart with numeric legend still was sorted.)

Reply:
@bfinzer I wish it was redundant, but it's not, at least not with things as they are. This part of the code is putting together the bar _covers_. Even though the actual bars would be sorted correctly without this, their associated covers would not be. We're iterating over `dataConfig.categorySetForAttrRole("legend")?.values` here which won't have the same order as `caseDataArray`. I thought maybe we could iterate over `caseGroups` instead and avoid the additional sort, but I couldn't get that to work. Maybe there's another option I'm overlooking?

Reply:
If nothing else, perhaps the compare function could be shared between the two implementations?

Reply:
@bfinzer @kswenson Does the latest commit seem reasonable?

Reply:
A detail I keep wondering about is the compare function name. Originally, I made it `compareNum` but that seemed too generic. I'm not sure `numericSortComparator` is quite right, though. Maybe something like `compareNumForSort` would be better?

Reply:
`numericSortComparator` seems fine.